test/test_qat.py

```diff
@@ -0,0 +1,126 @@
+from __future__ import absolute_import
+from __future__ import division
+from __future__ import print_function
+from __future__ import unicode_literals
+
+import torch
+from torch.nn import Conv2d, BatchNorm2d, ReLU
+from torch.nn._intrinsic.qat import ConvBn2d, ConvBnReLU2d
+from torch.quantization.QConfig import default_qat_qconfig
+from torch.nn import Parameter
+from common_utils import TestCase, run_tests
+from hypothesis import given
+from hypothesis import strategies as st
+from functools import reduce
+
+
+class IntrinsicQATModuleTest(TestCase):
+
+    @given(batch_size=st.integers(1, 3),
+           input_channels_per_group=st.sampled_from([2, 4, 5, 8, 16, 32]),
+           height=st.integers(10, 16),
+           width=st.integers(7, 14),
+           output_channels_per_group=st.sampled_from([2, 4, 5, 8, 16, 32]),
+           groups=st.integers(1, 3),
+           kernel_h=st.integers(1, 7),
+           kernel_w=st.integers(1, 7),
+           stride_h=st.integers(1, 2),
+           stride_w=st.integers(1, 2),
+           pad_h=st.integers(0, 2),
+           pad_w=st.integers(0, 2),
+           dilation=st.integers(1, 1),
+           use_bias=st.booleans(),
+           padding_mode=st.sampled_from(['zeros', 'circular']),
+           use_relu=st.booleans(),
+           eps=st.sampled_from([1e-5, 1e-4, 1e-3, 0.01, 0.1]),
+           momentum=st.sampled_from([0.1, 0.2, 0.3]))
+    def test_conv_bn_relu(
+            self,
+            batch_size,
+            input_channels_per_group,
+            height,
+            width,
+            output_channels_per_group,
+            groups,
+            kernel_h,
+            kernel_w,
+            stride_h,
+            stride_w,
+            pad_h,
+            pad_w,
+            dilation,
+            use_bias,
+            padding_mode,
+            use_relu,
+            eps,
+            momentum,
+    ):
+        freeze_bn = False
+        input_channels = input_channels_per_group * groups
+        output_channels = output_channels_per_group * groups
+        dilation_h = dilation_w = dilation
+
+        conv_op = Conv2d(
+            input_channels,
+            output_channels,
+            (kernel_h, kernel_w),
+            (stride_h, stride_w),
+            (pad_h, pad_w),
+            (dilation_h, dilation_w),
+            groups,
+            use_bias,
+            padding_mode
+        ).to(dtype=torch.float)
+        bn_op = BatchNorm2d(output_channels, eps, momentum).to(dtype=torch.float)
```

Comment:
Do we test if it is affine or not?

Reply:
we will enforce affine at the time of fusion.